Frontend-v1-Original/components/options/lib/useStakeData.ts

```
@@ -98,20 +106,49 @@ export function useTotalStaked() {
   const { data: tokenPrices } = useTokenPrices();
 
   return useQuery({
-    queryKey: ["totalStaked", tokenPrices, pair, totalSupply, pairData],
-    queryFn: () => getTotalStaked(pairData, totalSupply, tokenPrices),
-    enabled: !!pair && !!totalSupply && !!pairData && !!tokenPrices,
+    queryKey: [
+      "totalStaked",
+      tokenPrices,
+      pair,
+      totalSupply,
+      pairData,
+      stakedBalanceWithoutLock,
+      stakedBalanceWithLock,
+    ],
+    queryFn: () =>
+      getTotalStaked(
+        pairData,
+        totalSupply,
+        tokenPrices,
+        stakedBalanceWithoutLock,
+        stakedBalanceWithLock
+      ),
+    enabled:
+      !!pair &&
+      !!totalSupply &&
+      !!pairData &&
+      !!tokenPrices &&
+      !!stakedBalanceWithoutLock &&
+      !!stakedBalanceWithLock,
```

From what I know, the `queryFn` here will only run when `enabled` is true, so do we really still need to throw an error?

---

it's just for typescript in fact. otherwise you need to update function params in definiton and pass it with Non-null Assertion Operator (Postfix!).
Reference: [tkdodo](https://tkdodo.eu/blog/react-query-and-type-script#:~:text=I%27ve%20found%20the%20best%20way%20to%20go%20here%2C%20if%20you%20don%27t%20like%20the%20non%2Dnull%20assertion%20operator%2C%20is%20to%20accept%20that%20id%20can%20be%20undefined%20and%20reject%20the%20Promise%20in%20the%20queryFn.%20It%27s%20a%20bit%20of%20duplication%2C%20but%20it%27s%20also%20explicit%20and%20safe%3A)

---

Ok, I think I just came up with another hack.

```
queryFn: () => getTotalStaked(pairData!, totalSupply!, tokenPrices!),
```

---

> Technically, TypeScript is right, id is possibly undefined: the enabled option does not perform any type narrowing. Also, there are ways to bypass the enabled option, for example by calling the refetch method returned from useQuery. In that case, the id might really be undefined.
> 
> I've found the best way to go here, if you don't like the [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator), is to accept that id can be undefined and reject the Promise in the queryFn. It's a bit of duplication, but it's also explicit and safe.

---

Read that, but we don't call `refetch` in this case, do we?

---

what if we do?

---

i mean in any point in future

---

i guess i just like the way without non null assertion better